### PR TITLE
Add `isort` and `black` pre-commit hooks.

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary

This PR adds pre-commit hooks for `isort` and `black`.
After aru suggested using `isort` for imports and shiftinv pre-commits, I decided to create a PR for that.

The reason because I'm using those specifically versions is that both support python 3.10 (checked out in their changelogs)

Before merging, I would like a collaborator to answer this: should we add `isort` check in PRs/pushes?

## Checklist

- [x] If code changes were made then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `black .`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -1 +1,3 @@
-black>=21.9b0
+black>=21.11b0
+isort>=5.10.0
+pre-commit>=2.15.0
```
Comment (shiftinv) [inline requirements_dev.txt:3]:
```suggestion
black==21.11b1
isort==5.10.1
pre-commit~=2.15.0
```
`requirements_dev.txt` technically doesn't need to list the `black` and `isort`, as `pre-commit` installs its own environments and doesn't use the virtualenv, but I suppose we can keep them in there if someone wants to run one of the tools manually

**Review by shiftinv — CHANGES_REQUESTED** [review]
Looks good, the PR template would need updating as well.

The `pre-commit` hooks should be run and the changes should be committed in this PR (gotta be careful with star imports though), otherwise the next PR (whatever it may be) will change the entire code base as the contributing guide instructs contributors use `pre-commit`.
This will most likely result in merge conflicts in other PRs, though I suppose there's no way around that.

> Before merging, I would like a collaborator to answer this: should we add `isort` check in PRs/pushes?

We're already running `black` in workflows, it would make sense to replace that with a run of the `pre-commit` pipeline ([example](https://github.com/DisnakeDev/guide/blob/main/.github/workflows/mkdocs.yaml#L20-L21)).

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,10 @@
 [tool.black]
 line-length = 100
 target-version = ['py38', 'py39']
+
+[tool.isort]
+profile = "black"
+py_version = 38
+line_length = 100
```
Comment (shiftinv) [inline pyproject.toml:8]:
```suggestion
line_length = 100
combine_as_imports = true
```
I'd suggest using `combine_as_imports`, which avoids changes like this:
image

Comment (Victorsitou) [reply pyproject.toml:8]: Oh yeah, looks better, thank you, I'm going to apply this new configuration later.

**Review by shiftinv — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
-target-version = ['py38', 'py39']
+target-version = ["py38", "py39", "py310"]
+
+[tool.isort]
+profile = "black"
+py_version = 38
+line_length = 100
+combine_as_imports = true
+src_paths = ["disnake"]
+skip = ["examples", "tests"]
```
Comment (shiftinv) [inline pyproject.toml:11]:
```suggestion
extend_skip = ["examples", "tests"]
```

Comment (onerandomusername) [reply pyproject.toml]: why are we skipping these at all?

**Review by shiftinv — APPROVED** [review]

**Comment by onerandomusername** [comment]
I'll review this tomorrow\:tm:

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      - id: isort
+        args: ["--profile", "black", "--extend-skip", "examples"]
+        name: Running isort in all files.
+      - id: isort
+        args: ["--profile", "black", "--thirdparty", "disnake"]
+        name: Running isort in examples.
+        files: ^examples/
+
+  - repo: https://github.com/psf/black
+    rev: 21.11b1
```
Comment (onerandomusername) [inline .pre-commit-config.yaml:15]: should we consider updating black before merging?

Comment (Victorsitou) [reply .pre-commit-config.yaml:15]: Yeah 21.12 looks good

File: requirements_dev.txt [hunk 5]
```diff
@@ -1 +1,3 @@
-black>=21.9b0
+black==21.11b1
```
Comment (onerandomusername) [inline requirements_dev.txt:1]: black also needs to be changed here

**Review by onerandomusername — APPROVED** [review]
it looks good enough for me

**Comment by onerandomusername** [comment]
Linking the comment that started this pull for posterity: https://github.com/DisnakeDev/disnake/pull/165#issuecomment-979477043

**Pull request merged**